rfcs/0000-type-stability.md

```diff
@@ -0,0 +1,499 @@
+- TypeScript releases roughly quarterly. There are no LTS releases for TypeScript.
+
+These map reasonably well to each other, and this correspondence provides a helpful basis for recommending a version support policy.
+
+Addons should generally support the TypeScript versions current during the lifetime of the Ember LTS versions they support. As noted above, dropping a supported TypeScript version is a breaking change, and it is helpful to minimize the number of major versions of an addon in use.
+
+#### Supporting new versions
+
+The Typed Ember team shall normally make a recommendation within a week of the release of a new TypeScript version whether it should be adopted as a supported version by the community.
+
+Some TypeScript releases have bugs which last throughout the life of the release and are fixed only in the following release. This is not uncommon; the same is true of ember-source and ember-data. Critically, however, the role played by TypeScript in the development pipeline means that users—including addon authors—can safely wait for later releases. This is not *common*, but has happened enough times in recent TypeScript history (including 2.9–3.1, 3.5, and 3.8) that a recommended support matrix will minimize churn and pain in the ecosystem.
+
+Addon authors should *prefer* to follow guidance from the Typed Ember team about supported TypeScript versions. Addon authors *may* opt into supporting non-recommended versions, but the Typed Ember team will not commit to helping support TypeScript issues and bugs for addons which do so. (See [<b>Documenting supported versions</b>](#documenting-supported-versions) below.)
+
+Adding a new supported version does *not* require an addon to create a new release, though it should be added to addons’ test suites and their  documentation about supported versions.
+
+#### Dropping TypeScript versions
+
+As discussed above in [Changes to types: Dropping support for previously-supported versions](#dropping-support-for-previously-supported-versions), dropping support for previously-supported versions of TypeScript is always a breaking change. As such, it should always occur in a major version. As such, and given the constraints of Ember CLI’s version resolution today (where only one major version will be resolved successfully), addon authors should prefer to drop older supported TypeScript versions relatively infrequently. Optimally, addon authors should drop TypeScript versions at the same time as they are already making *other* breaking changes of a similar sort, like dropping support for Node or Ember versions when their LTS period ends.
+
+#### Documenting supported versions
+
+Addons should document their currently supported TypeScript versions in their README. Typed Ember shall publish a repository with a README listing the specified versions, which will be home to a package `@typed-ember/supported-versions`, whose `index.js` is an array of version numbers, which can be used programmatically:
```

Comment:
What use case did you have in mind for programmatically consuming this information?

Reply:
I was thinking that the tooling we build could use it to generate the desired configuration in line with our guidance automatically. That said… it could just live *in the tool*, and that's probably better?

Reply:
Removed this in e5b4311.